cnxarchive/tests/test_views.py

```diff
@@ -1444,6 +1444,20 @@ def test_exports_without_version(self):
         with self.assertRaises(IdentHashMissingVersion) as cm:
             get_export(self.request)
 
+    def test_get_extra_404(self):
+        id = 'b771c6fc-34f0-11e7-ad77-e3343f783f02'
+        version = '1.1'
+
+        # Build the request
+        self.request.matchdict = {'ident_hash': '{}@{}'.format(id, version)}
+        self.request.matched_route = mock.Mock()
+        self.request.matched_route.name = 'content-extras'
+
+        from ..views import get_extra
+
+        self.assertRaises(httpexceptions.HTTPNotFound,
```

Comment:
@reedstrm For future reference, it's often times easiest to use the `with` statement when using the `assertRaises` because it makes it far easier to step into the function in question from a `pdb` prompt.